homeassistant/components/media_player/songpal.py

```diff
@@ -114,13 +122,79 @@ def __init__(self, name, endpoint):
         self._volume = 0
         self._is_muted = False
 
-        self._sources = []
+        self._active_source = None
+        self._sources = {}
+
+    @property
+    def should_poll(self):
+        """Return True if the device should be polled."""
+        return self._poll
```

Comment:
Should we keep the attribute?

Reply:
I'd prefer keeping it around for the time being, if someone comes up with a broken setup it makes it easy to check if the polling variant would work in that case.